### PR TITLE
test-analytics: Add tables and view for ci-failures, Store known issues

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        build_date,
+        mz_version,
+        commit_hash,
+        build_step_key,
+        test_retry_count,
+        occurrence_count,
+        build_id,
+        build_job_id
+    FROM v_build_annotation_error
+    GROUP BY
```
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql:26]: I don't get the aggregation here? Do we group by all fields except `content`? Why? And don't we need to use `sum(occurrence_count)`?

File: misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+--
+-- Use of this software is governed by the Business Source License
+-- included in the LICENSE file at the root of this repository.
+--
+-- As of the Change Date specified in that file, in accordance with
+-- the Business Source License, use of this software will be governed
+-- by the Apache License, Version 2.0.
+
+CREATE OR REPLACE MATERIALIZED VIEW mv_ci_failures AS
+    SELECT
```
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql:11]: Specify the cluster?

File: test/cluster/mzcompose.py [hunk 3]
```diff
@@ -223,7 +223,7 @@ def workflow_test_github_12251(c: Composition) -> None:
         SET statement_timeout = '1 s';
         CREATE TABLE IF NOT EXISTS log_table (f1 TEXT);
         CREATE TABLE IF NOT EXISTS panic_table (f1 TEXT);
-        INSERT INTO panic_table VALUES ('forced panic');
+        INSERT INTO panic_table VALUES ('forcedasdsad panic');
```
Comment (nrainer-materialize) [inline test/cluster/mzcompose.py:226]: 😀

**Review by nrainer-materialize — CHANGES_REQUESTED** [review]

File: misc/python/materialize/test_analytics/setup/tables/40-issues.sql [hunk 4]
```diff
@@ -0,0 +1,10 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Use of this software is governed by the Business Source License
+-- included in the LICENSE file at the root of this repository.
+--
+-- As of the Change Date specified in that file, in accordance with
+-- the Business Source License, use of this software will be governed
+-- by the Apache License, Version 2.0.
+
+CREATE TABLE issues (issue TEXT NOT NULL, title TEXT NOT NULL, ci_regexp TEXT, state TEXT);
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/test_analytics/setup/tables/40-issues.sql:10]: I'm wondering if we should actually have multiple ci-regexp. Maybe just use `(foo|bar)` instead

File: misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql (hunk 1 above)
Comment (def-) [reply misc/python/materialize/test_analytics/setup/views/30-ci-failures.sql:26]: Yes, I just wanted to catch the (relatively common) case where there are multiple errors in the same run.

File: misc/python/materialize/test_analytics/data/known_issues/known_issues_storage.py [hunk 5]
```diff
@@ -22,21 +22,21 @@ def add_issue(
         )
         sql_statements = [
             f"""
-            UPDATE issues
+            UPDATE issue
             SET title = {mz_sql_util.as_sanitized_literal(issue.info["title"])},
                 ci_regexp = {mz_sql_util.as_sanitized_literal(issue.regex.pattern.decode("utf-8"))},
                 state = {mz_sql_util.as_sanitized_literal(issue.info["state"])}
-            WHERE issue = {mz_sql_util.as_sanitized_literal(issue_str)}
+            WHERE issue_id = {mz_sql_util.as_sanitized_literal(issue_str)}
```
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/data/known_issues/known_issues_storage.py:29]: Column name mismatch: `issue_id` vs. `issue_reference` in table definition

File: misc/python/materialize/test_analytics/setup/views/100-data-integrity.sql [hunk 6]
```diff
@@ -74,4 +74,8 @@ CREATE OR REPLACE VIEW v_data_integrity (table_name, own_item_key, referenced_it
     SELECT 'config', 'config', NULL, 'more than one config entry exists'
     FROM config
     HAVING count(*) > 1
+    UNION
+    SELECT 'issue', issue_reference, NULL, 'more than one issue entry exists'
```
Comment (nrainer-materialize) [inline misc/python/materialize/test_analytics/setup/views/100-data-integrity.sql:78]: I think you need a `GROUP BY` on `issue_reference`

**Comment by def-** [comment]
Added a bunch of fixes on top, ready for review again

**Review by nrainer-materialize — APPROVED** [review]
Thanks for addressing the comments.

**Pull request merged**